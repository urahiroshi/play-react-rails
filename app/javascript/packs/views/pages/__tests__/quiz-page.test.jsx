/* eslint react/prop-types: 0 */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import QuizPage from '../quiz-page.jsx'

const createComponent = ({ questions }) => (
  <QuizPage questions={questions} startGetQuestions={() => {}} />
)

describe('when questions are undefined', () => {
  it('should show empty element', () => {
    const wrapper = shallow(createComponent({ questions: undefined }))
    expect(wrapper.matchesElement(<div />)).toBe(true)
  })
})

describe('when questions are empty', () => {
  it('should show no questions message', () => {
    const wrapper = shallow(createComponent({ questions: [] }))
    expect(wrapper.contains('There are no questions!')).toBe(true)
  })
})

describe('when questions are not empty', () => {
  const QUESTIONS = [
    {
      id: 1,
      content: 'hogehoge',
      answer: 'hoge',
    }, {
      id: 2,
      content: 'fugafuga',
      answer: 'fuga',
    },
  ]

  it('should show quiz page', () => {
    const wrapper = shallow(createComponent({ questions: QUESTIONS }))
    expect(wrapper.is('.quiz-page')).toBe(true)
  })

  it('should show snapshot', () => {
    const wrapper = shallow(createComponent({ questions: QUESTIONS }))
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
