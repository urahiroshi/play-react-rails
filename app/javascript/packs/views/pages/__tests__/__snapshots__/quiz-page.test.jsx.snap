// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when questions are not empty should show snapshot 1`] = `
<div
  className="quiz-page center-block"
>
  <QuizBody
    moveNextPage={[Function]}
    page={1}
    question={
      Object {
        "answer": "hoge",
        "content": "hogehoge",
        "id": 1,
      }
    }
  />
  <PagingButton
    className="quiz-prev"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-chevron-left"
    />
  </PagingButton>
  <PagingButton
    className="quiz-next"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-chevron-right"
    />
  </PagingButton>
</div>
`;
